src/frontend/js/components/PaymentButton/index.spec.tsx

```diff
@@ -0,0 +1,472 @@
+    const handleSuccess = jest.fn();
+
+    fetchMock
+      .post('https://joanie.test/api/orders/', order)
+      .get(`https://joanie.test/api/orders/${order.id}/`, {
+        ...order,
+        state: OrderState.PENDING,
+      })
+      .post(`https://joanie.test/api/orders/${order.id}/abort/`, 200);
+
+    const { clientState } = PersistedClientFactory({
+      queries: [QueryStateFactory('user', { data: { username: 'John Doe' } })],
+    });
+    const client = createQueryClient();
+    hydrate(client, clientState);
+
+    render(
+      <Wrapper client={client}>
+        <PaymentButton
+          billingAddress={billingAddress}
+          creditCard={creditCard.id}
+          product={product}
+          onSuccess={handleSuccess}
+        />
+      </Wrapper>,
+    );
+
+    const $button = screen.getByRole('button', {
+      name: `Pay in one click ${formatPrice(product.price, product.price_currency)}`,
+    }) as HTMLButtonElement;
+
+    // - As all information are provided, payment button should not be disabled.
+    expect($button.disabled).toBe(false);
+
+    // - User clicks on pay button
+    await act(async () => {
+      fireEvent.click($button);
+    });
+
+    // - Route to create order should have been called
+    // - Furthermore, as payment succeeded immediately, order should have been refetched
+    expect(fetchMock.calls()).toHaveLength(2);
+    expect(fetchMock.calls()[0][0]).toBe('https://joanie.test/api/orders/');
+    expect(JSON.parse(fetchMock.calls()[0][1]!.body as string)).toEqual({
+      billing_address: billingAddress,
+      credit_card_id: creditCard.id,
+      course: '00000',
+      product: product.id,
+    });
+    expect(fetchMock.calls()[1][0]).toBe(`https://joanie.test/api/orders/${order.id}/`);
+
+    // - Spinner should be displayed
+    screen.getByText('Payment in progress');
+
+    // - Payment interface should not be displayed
+    expect(screen.queryByText('Payment interface component')).toBeNull();
+
+    fetchMock.resetHistory();
+    // - Wait until order has been polled 29 times.
+    await waitFor(
+      async () => {
+        expect(fetchMock.calls()).toHaveLength(PAYMENT_SETTINGS.pollLimit - 1);
+      },
+      {
+        timeout: PAYMENT_SETTINGS.pollLimit * 1000,
+        interval: 5,
+      },
+    );
+
+    // - This round should be the last after which the order should be aborted
+    await act(async () => {
+      jest.runOnlyPendingTimers();
+    });
+    expect(fetchMock.calls()).toHaveLength(PAYMENT_SETTINGS.pollLimit);
+    expect(fetchMock.lastUrl()).toBe(`https://joanie.test/api/orders/${order.id}/abort/`);
+    expect(JSON.parse(fetchMock.lastOptions()!.body!.toString())).toEqual({
+      payment_id: order.payment_info.payment_id,
+    });
+
+    // - An error message should be displayed
+    screen.getByText('An error occurred during payment. Please retry later.');
+    await act(async () => {
+      jest.runAllTimers();
```

Comment:
Please can you explain why you are doing this at the end of the test?
```suggestion
    await act(async () => {
      jest.runAllTimers();
```

Reply:
This is a residue of a vicious bug